bolt-modules/boltlib/lib/puppet/datatypes/resourceinstance.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+Puppet::DataTypes.create_type('ResourceInstance') do
+  interface <<-PUPPET
+    attributes => {
+      'target'        => Target,
+      'type'          => String[1],
```

Comment:
Should this also support `Type[Resource]` like get_resources? https://github.com/puppetlabs/bolt/blob/master/bolt-modules/boltlib/lib/puppet/functions/get_resources.rb#L22

Reply:
Updated to include `Type[Resource]` as a type.